pkg/model/resource/options_test.go

```diff
@@ -73,24 +74,37 @@ var _ = Describe("Resource Options", func() {
 			Expect(options.Validate().Error()).To(ContainSubstring("kind cannot be empty"))
 		})
 
-		It("should fail if the Kind is not pascal cased", func() {
-			// Base case
-			options := &Options{Group: "crew", Version: "v1", Kind: "FirstMate"}
-			Expect(options.Validate()).To(Succeed())
-
-			// Can't detect this case :(
-			options = &Options{Group: "crew", Version: "v1", Kind: "Firstmate"}
-			Expect(options.Validate()).To(Succeed())
+		DescribeTable("valid Kind values - according to core Kubernetes",
+			func(kind string) {
+				options := &Options{Group: "crew", Kind: kind, Version: "v1"}
+				Expect(options.Validate()).To(Succeed())
+			},
+			Entry("camelcase", "FirstMate"),
+			Entry("more than one caps at the start", "FIRSTMate"),
+		)
+
+		It("should fail if Kind is too long", func() {
+			var kind string
+
+			for i := 0; i < 65; i++ {
+				kind += "a"
+			}
+
+			options := &Options{Group: "crew", Kind: kind, Version: "v1"}
+			err := options.Validate()
+			Expect(err).To(MatchError(ContainSubstring("must be no more than 63 characters")))
+		})
 
-			options = &Options{Group: "crew", Version: "v1", Kind: "firstMate"}
-			Expect(options.Validate()).NotTo(Succeed())
-			Expect(options.Validate().Error()).To(ContainSubstring(
-				`kind must be PascalCase (expected FirstMate was firstMate)`))
+		It("should fail if Kind is invalid, according to core Kubernetes", func() {
```

Comment:
```suggestion
		It("should fail if Kind has whitespaces", func() {
```

Reply:
Good point; I've added a small collection of `Kind`s that would be considered invalid instead-of which whitespace is one. My reasoning here is that this test also serves as documentation for future us :) (or other humans that work on this code) on what logic we decided more broadly.